### PR TITLE
preflight script constants

### PR DESCRIPTION
**Pull request by davidsbailey**
Starts [ACQ-169](https://codedotorg.atlassian.net/browse/ACQ-169). Depends on https://github.com/code-dot-org/code-dot-org/pull/48625.

The goal is to move script_constants.rb to dashboard. This is a "preflight" PR that attempts to make some changes ahead of time, to make the subsequent PR smaller and less risky. This work is a step toward eliminating [manual work in script_constants.rb](https://docs.google.com/document/d/11tbt6RiygkRtwv1m2p8Bt0opOQQAzJF695Gzd-V3T3s/edit#heading=h.bkemgfu9nao), as well as toward separating pegasus from dashboard.

The only user-visible change in this PR is that `code.org/congrats/<coursename>` is now indiscriminate, in the sense that it will redirect any coursename to `studio.code.org/congrats?s=...`, rather than just CSF courses. since there are no user flows we know of that would send users to code.org/congrats/foo for anything but a CSF course, this seems like a reasonable sacrifice.

If you do go to code.org/congrats/foo, you get to https://studio.code.org/congrats/s=Zm9v, which 404s if the course is not a HOC or CSF course. If users start running into this, a simple fix would be to redirect to /congrats (no s param) when the s param is invalid.

## Testing story

existing test coverage

## Follow-up work

actually move script_constants.rb into dashboard

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -343,7 +343,7 @@ def all_scripts
 
     def family_names
       Rails.cache.fetch('script/family_names', force: !Script.should_cache?) do
-        (CourseVersion.course_offering_keys('Script') + ScriptConstants::FAMILY_NAMES).uniq.sort
+        CourseVersion.course_offering_keys('Script').uniq.sort
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb:346]: reminder: potential merge conflict with Script -> Unit rename PR

**Review by megcrenshaw — APPROVED** [review]
I spot-checked some of the constants deleted and it looks good. Did you have a consistent strategy to decide what to delete and what to keep?

**Comment by davidsbailey** [comment]
> I spot-checked some of the constants deleted and it looks good. Did you have a consistent strategy to decide what to delete and what to keep?

Yes - I tried to remove every constant that wasn't being used.

**Comment by dmcavoy** [comment]
❤️ this PR so much!

File: lib/cdo/script_constants.rb [hunk 2]
```diff
@@ -86,9 +88,6 @@ module ScriptConstants
       EXPLORE_DATA_1_2021_NAME = 'explore-data-1-2021'.freeze,
       SPELLING_BEE_2021_NAME = 'spelling-bee-2021'.freeze
     ],
-    aiml_2021: [
-      AIML_2021_NAME = 'aiml-2021'.freeze,
```
Comment (bethanyaconnor) [inline lib/cdo/script_constants.rb:90]: Talking to @dju90, it sounds like AIML will be added to the translation pipeline sometime soon so maybe we should keep this constant?

**Review by bethanyaconnor — APPROVED** [review]

File: lib/cdo/script_constants.rb (hunk 2 above)
Comment (davidsbailey) [reply lib/cdo/script_constants.rb:90]: added!

**Pull request merged**